clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1419,6 +1419,28 @@ is not specified.
   }];
 }
 
+def ExplicitInitDocs : Documentation {
+  let Category = DocCatField;
+  let Content = [{
+The ``clang::explicit_init`` attribute indicates that the field must be
+initialized explicitly by the caller when the class is constructed.
```

Comment:
Maybe document that the caveats here?
I could see us exploring the extension of this analysis to make sure the "explicit_init" fields are all written before they're read, but that's not what's happening today IIUC.

Another caveat, is that the attribute is ignored on non-aggregate classes.

```cpp
struct NonAgg {

  NonAgg()  {}
  int x [[clang::explicit_init]]; // attribute ignored.
}
```

And maybe also mention that these warnings do not aim to comprehensively prevent uses of uninitialized memory, but are, rather, supporting tools for aiding to write code in certain code style if people choose to (e.g. using it for parameter objects in combination with C++20 designated initializer).

Reply:
Thanks! LG!

> we might want to forbid its usage on non-aggregates entirely.

I guess we could simply warn if the class in which the field is used is non-aggregate, but that entails the problems that were raised with the changed semantics of what's aggregate between C++17 and C++20 (especially if people use `Werror`).
I guess having a warning that is off by default is a no-brainer, it clearly adds value **if** people want it. But it might also happen that nobody uses it and the work for it is wasted.

Or we could simply document that the attribute is ignored for non-aggregate types in the documentation.